x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_overview.test.tsx

```diff
@@ -72,6 +72,7 @@ describe('EnginesOverview', () => {
       const wrapper = shallow(<EnginesOverview />);
 
       expect(wrapper.find(EnginesTable)).toHaveLength(1);
+      expect(wrapper.find('[data-test-subj="appSearchEnginesCreateEngineButton"]')).toHaveLength(1);
```

Comment:
Rather than simply testing the existence of the button, we need to actually simulate clicking the button itself in order to get functional coverage on its `onClick` handler. I would put a new `it()` test block below this one that looks like this:

```js
import { setMockValues, setMockActions, mockKibanaValues, rerender } from '../../../__mocks__';

// ... etc

it('renders a create engine button which takes users to the create engine page', () => {
  const { navigateToUrl } = mockKibanaValues;
  const wrapper = shallow(<EnginesOverview />);

  wrapper.find('[data-test-subj="appSearchEnginesCreateEngineButton"]').simulate('click');

  expect(navigateToUrl).toHaveBeenCalledWith('/create_engine');
});
```

We should remove the `toHaveLength` check after adding this test, because that's implicit in the .find/.simulate action (Jest will throw if the button doesn't actually exist and we try to simulate a click on it).